Add Gateway For Mesh section to the GEP template (#3577)
kubernetes-sigs/gateway-api: geps/gep-696/index.md

-- e.g. owner messes up the permissions, accidental deletion)

## API
(... details, can point to PR with changes)

### Gateway for Ingress (North/South)
(Include API details for North/South usecases)

### Gateway For Mesh (East/West)

> (Include East/West API considerations, examples, and if differnet - apis)

Reviewer: ```suggestion
(Include East/West API considerations, examples, and if different - APIs)
```

## Conformance Details

(from https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-2162/index.md#standardize-features-and-conformance-tests-names)

#### Feature Names

> (Does it require seperate feature name for mesh? Please add it if necessary)

Reviewer: ```suggestion
(Does it require separate feature(s) for mesh? Please add them if necessary)
```

In any case, the conformance tests file names should be meaningful and easy to understand.

> (MAKE sure to also include conformance tests that covers mesh)

Reviewer: ```suggestion
(Make sure to also include conformance tests that cover mesh)
```
